04week/mastermind.js

```diff
@@ -22,19 +22,67 @@ function generateSolution() {
     const randomIndex = getRandomInt(0, letters.length);
     solution += letters[randomIndex];
   }
+  // console.log(solution);
+  return solution
 }
 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(guess) {
   // your code here
+
+  let solutionArray = solution.split("")
+  let guessArray = guess.split("")
+  let correctLetterLocations = 0
+  let correctLetters = 0
+  let hint = 0
+  // console.log(solution.length)
+
+  for (let i = 0; i < solution.length; i++){
+      if (guessArray[i] == solutionArray[i]){
+      correctLetterLocations ++;
+      solutionArray[i] = null;
+      // console.log(solutionArray)
+      }
+    }
+    for (let x = 0; x<= solution.length; x++){
+      if (solutionArray.indexOf(guessArray[x]) > -1){
+        correctLetters++
+        solutionArray[x] = null;
```

Comment:
you need to null out the solution Array at the position the letter exists in the solution array, not in the guessArray